src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,47 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int USER_MINIMAL_AGE = 18;
+    private static final int MINIMAL_PASSWORD_LENGTH = 6;
+    private static final int MINIMAL_LOGIN_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("User can't be null");
+        }
+        if (isUserLoginAlreadyExist(user)) {
+            throw new RegistrationException("User with this login already exist");
+        }
+        if (isMinimalUserAgeNotMet(user)) {
+            throw new RegistrationException("Minimal age requirement isn't met");
+        }
+        if (isStringLengthNotMet(user.getLogin(), MINIMAL_LOGIN_LENGTH)) {
+            throw new RegistrationException("Minimal login length requirement isn't met");
+        }
+        if (isStringLengthNotMet(user.getPassword(), MINIMAL_PASSWORD_LENGTH)) {
+            throw new RegistrationException("Minimal password length requirement isn't met");
+        }
```

Comment:
Cool, but what do you think about extracting entire validation stuff to the separate methods? Like this pseudocode
`void validate(user) throws Exception`

Thus we can just make a method call in the `register` and proceed next

Reply:
change structure, mb fixed )